### DISCUSSION TITLE
Implementing Assays As Children Of Studies

### DISCUSSION THREAD

**Discussion by ptth222** [MoseleyBioinformaticsLab/MESSES#34] [General]
We somewhat briefly mentioned this in this ISA Protocol Parameters and Components #32 discussion and talked about it in lab meeting a bit, but now that I have added some more code to the pre-directive section for ISA I think this is a better idea now. 

Basically, assays would go in the study table, but would require a study parent. This gives a convenient way to group the assay information into one place for directives and user override. You had mentioned in the meeting that we should add a "type" field that should be "study" or "assay" as well, but I have some questions about other details and an issue. 

1. In our example data we used "type" as a field already. Ex. 'type': 'preliminary data'. This is isn't actually used by anything, but I will have to redo/recheck the examples. 
2. Should we validate study inheritance more strictly? For example, if a study is third in a lineage (is a child of an assay) that doesn't really make any sense, so should we only allow 1 level of inheritance?
3. Should "type" be required or only for assays? In order to be more backward compatible I think we can validate so that if there is no "parent_id" we assume it is a study and don't require a "type" field, but if there is a "parent_id" we require a "type" field that says "assay". The "type" field is a little redundant altogether, but does offer some advantages.

**Comment by hunter-moseley** [comment]
1) We can use another fieldname besides "type", if this one is taken. Maybe
"record_type", "entry_type", or "isa_record_type".

2) Can an assay be the parent of a derivative assay? If so, then maybe do
not limit the level.

3) I was suggesting a "type"field for readability purposes.  Would not
require it.  If a study could be a parent of another non-assay study, then
the field must be used to prevent the interpretation of the study with the
parent_id as being an assay.


On Wed, Nov 8, 2023 at 4:38 PM ptth222 ***@***.***> wrote:

> We somewhat briefly mentioned this in this ISA Protocol Parameters and
> Components #32
> <https://github.com/MoseleyBioinformaticsLab/MESSES/discussions/32>
> discussion and talked about it in lab meeting a bit, but now that I have
> added some more code to the pre-directive section for ISA I think this is a
> better idea now.
>
> Basically, assays would go in the study table, but would require a study
> parent. This gives a convenient way to group the assay information into one
> place for directives and user override. You had mentioned in the meeting
> that we should add a "type" field that should be "study" or "assay" as
> well, but I have some questions about other details and an issue.
>
>    1. In our example data we used "type" as a field already. Ex. 'type':
>    'preliminary data'. This is isn't actually used by anything, but I will
>    have to redo/recheck the examples.
>    2. Should we validate study inheritance more strictly? For example, if
>    a study is third in a lineage (is a child of an assay) that doesn't really
>    make any sense, so should we only allow 1 level of inheritance?
>    3. Should "type" be required or only for assays? In order to be more
>    backward compatible I think we can validate so that if there is no
>    "parent_id" we assume it is a study and don't require a "type" field, but
>    if there is a "parent_id" we require a "type" field that says "assay". The
>    "type" field is a little redundant altogether, but does offer some
>    advantages.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/MoseleyBioinformaticsLab/MESSES/discussions/34>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADEP7B5AJNTXOMRZEBU3P3DYDP3VVAVCNFSM6AAAAAA7DR3ZBGVHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZVHAZDSNZXGY>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Hunter Moseley, Ph.D. -- Univ. of Kentucky
Professor, Dept. of Molec. & Cell. Biochemistry / Markey Cancer Center
/ Institute for Biomedical Informatics / UK Superfund Research Center
 Not just a scientist, but a fencer as well.
   My foil is sharp, but my mind sharper still.
---------------------------------------------------------------
Email: ***@***.*** (work)   ***@***.***
(personal)
Phone: 859-218-2964 (office)   859-218-2965 (lab)   859-257-7715 (fax)
Web: http://bioinformatics.cesb.uky.edu/
Address: CC434 Roach Building, 800 Rose Street, Lexington, KY 40536-0093

**Comment by ptth222** [comment]
We discussed this in a meeting.
 1. "type" should be used since we do something similar in the entity table for "subject" and "sample".
 2. For ISA the additional restrictions on inheritance make sense, but we don't want to limit things for the future, so I will add extra validation for ISA only.
 3. Same as 2. "type" won't be required, but will have extra checks for ISA only.
